src/openrct2/paint/VirtualFloor.cpp

```diff
@@ -195,7 +196,7 @@ bool VirtualFloorTileIsFloor(const CoordsXY& loc)
         return true;
     }
 
-    if (gMapSelectFlags & MAP_SELECT_FLAG_ENABLE_CONSTRUCT)
+    if (gMapSelectFlags & MAP_SELECT_FLAG_ENABLE_CONSTRUCT && !isToolActive(WindowClass::Footpath))
```

Comment:
This seems a bit hacky. Any particular reason this isn't at the calling side?

Reply:
That was more of a test whether I was on the right track. Pushed it rather out of habit.

I have now changed it to simply fail early in case `MAP_SELECT_FLAG_ENABLE` is set, but the current tile is not close to the selection. That appears to do the trick. (Checked with ride, footpath, small scenery and large scenery, and all combinations of the last 3 and an open ride construction window.)

Will do the clean up tomorrow, after I get some sleep.

Do you prefer keeping both PRs separate or shall I merge this one into #23502 ?

Reply:
We can merge them separately at this point. Different issues as well, after all.